### PR TITLE
monopoly game is under progress

### PR DESCRIPTION
**Pull request by LouiseMedova**

### CODE REVIEW

File: monopoly/io/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    Silver,
+    Gold,
+}
+
+impl Gear {
+    pub fn upgrade(&self) -> Self {
+        match *self {
+            Self::Platinum => Self::Silver,
+            Self::Silver => Self::Gold,
+            Self::Gold => Self::Gold,
```
Comment (azoyan) [inline monopoly/io/src/lib.rs]: Usually upgrade mean that metal should changes to more prestigious metal, for example, silver to gold, gold to platinum, platinum to platinum. Maybe I'm misunderstanding something?

Comment (LouiseMedova) [reply monopoly/io/src/lib.rs]: Yes exactly, I thought Platinum is worse than Gold :)

Comment (azoyan) [reply monopoly/io/src/lib.rs]: We can use more natural and familiar categories: bronze, silver, gold.

File: monopoly/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+                    .get_mut(&player)
+                    .expect("Cant be None: Get Player");
+                debug!("PLAYER_INFO {:?}", player_info.clone());
+
+                // if a player is in jail we don't throw rolls for him
+                let position = if player_info.in_jail {
+                    player_info.position
+                } else {
+                    let (r1, r2) = get_rolls();
+                    debug!("ROOLS {:?} {:?}", r1, r2);
```
Comment (azoyan) [inline monopoly/src/lib.rs]:
```Rust
debug!("ROLLS {:?} {:?}", r1, r2);
```

File: monopoly/src/utils.rs [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+            }
+        } else {
+            debug!("PENALTY: TRY TO SELL FREE PROPERTY");
+            player_info.penalty += 1;
+            return Err(());
+        };
+    }
+
+    for property in properties_for_sale {
+        let (_, price, _) = properties.get(property).expect("Properies: Cant be None");
```
Comment (azoyan) [inline monopoly/src/utils.rs]:
```Rust
let (_, price, _) = properties.get(property).expect("Properties: Cant be None");
```

File: monopoly/src/strategic_actions.rs [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+            };
+        }
+
+        let (r1, r2) = get_rolls();
+        if r1 == r2 {
+            player_info.in_jail = false;
+            player_info.position = r1 + r2;
+        } else if pay_fine {
+            if player_info.balance < FINE {
+                debug!("PENALTY: NOT ENOUGGH BALANCE");
```
Comment (azoyan) [inline monopoly/src/strategic_actions.rs]:
```Rust
debug!("PENALTY: NOT ENOUGH BALANCE");
```

File: monopoly/src/strategic_actions.rs [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+            Ok(player_info) => player_info,
+            Err(_) => {
+                reply_strategic_error();
+                return;
+            }
+        };
+        let position = player_info.position;
+        // if a player tries to buy position that has already been bought
+        if self.ownership.contains_key(&position) {
+            debug!("PENALTY: THAT CELL IS ALREDY BOUGHT");
```
Comment (azoyan) [inline monopoly/src/strategic_actions.rs]:
```Rust
debug!("PENALTY: THAT CELL IS ALREADY BOUGHT");
```

**Pull request merged**